federation/pkg/federation-controller/service/endpoint_helper_test.go

```diff
@@ -17,11 +17,32 @@ limitations under the License.
 package service
 
 import (
+	"sync"
 	"testing"
 
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns" // Only for unit testing purposes.
 	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/sets"
 )
 
+var fakeDns, _ = clouddns.NewFakeInterface()
+var fakeDnsZones, _ = fakeDns.Zones()
+
+var fakeServiceController = ServiceController{
+	dns:              fakeDns,
+	zones:            fakeDnsZones,
+	federationClient: nil,
+	serviceCache:     &serviceCache{fedServiceMap: make(map[string]*cachedService)},
+	clusterCache: &clusterClientCache{
+		rwlock:    sync.Mutex{},
+		clientMap: make(map[string]*clusterCache),
+	},
+	eventBroadcaster: nil, /* broadcaster*/
+	eventRecorder:    nil, /*recorder**/
+	queue:            nil, /*workqueue.New()*/
```

Comment:
Why set these (and `federationClient` above) to `nil`? That's the default zero-value anyway. Removing them makes this more concise.